.tekton/build_and_trigger.yaml

```diff
@@ -0,0 +1,109 @@
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: build-and-trigger
+  annotations:
+    pipelinesascode.tekton.dev/max-keep-runs: "3"
+    pipelinesascode.tekton.dev/on-cel-expression: |
+            event == "push" || event == "pull_request"
+    pipelinesascode.tekton.dev/task: "[buildah, git-clone]"
+spec:
+  params:
+    - name: output-image
+      value: quay.io/redhat-appstudio/clair-in-ci:to_test
+    - name: builder-image
+      value: registry.access.redhat.com/ubi9/buildah:9.0.0-19@sha256:c8b1d312815452964885680fc5bc8d99b3bfe9b6961228c71a09c72ca8e915eb
+    - name: dockerfile
+      value: Dockerfile
+    - name: revision
+      value: "{{ revision }}"
+    - name: target_branch
+      value: "{{ target_branch }}"
+    - name: source_branch
+      value: "{{ source_branch }}"
+    - name: repo_url
+      value: "{{ repo_url }}"
+  pipelineSpec:
+    params:
+      - name: output-image
+      - name: builder-image
+      - name: dockerfile
+      - name: revision
+      - name: target_branch
+      - name: source_branch
+      - name: repo_url
+    workspaces:
+      - name: workspace
+    tasks:
+      - name: fetch-repo
+        workspaces:
+          - name: output
+            workspace: workspace
+        taskRef:
+          name: git-clone
+        params:
+          - name: url
+            value: "$(params.repo_url)"
+          - name: revision
+            value: "$(params.revision)"
```

Comment:
IIUC, with event==daily, you want to build from main branch of main repo, not from daily branch.}

So this must be somehow parametrized

Reply:
Hint: I'd create init task which would output:
* event
* revision (HEAD for main branch, I dunno how to make it into SHA revision w/o cloning first)

Reply:
on the other hand, we don't need SHA revision for image_tag, we can just use timestamps for daily and main, and PR number for pull-request builds. So we can try revision HEAD just for clonning.

Using timestamp has another benefit, we can drop logic for retagging last image as previous, becasue timestamps unlike SHAs will be always unique for nightly builds.

So init task can return:
* event
* revision (HEAD for daily)
* image_tag (PR number for pullrequest event, timestamp for push and daily)

Reply:
Daily is build from main: 
https://github.com/redhat-appstudio/clair-in-ci-db/pull/1/files#diff-8ad847f60ffa7472caec815fbd7c8b6f5f185f9e3472956678100636bbee3c37R17
I will rename those parameters so its more clear > source_branch & target_branch.

Reply:
Source and target is set from the PR that will be created periodically, setting source branch as main will do this trick.

Reply:
ah I see, it deserves comment in code then